dotnet/Gherkin/Parser.cs

```diff
@@ -89,7 +101,12 @@ protected class ParserContext
 
         public T Parse(ITokenScanner tokenScanner)
         {
-            return Parse(tokenScanner, new TokenMatcher(), new AstBuilder<T>());
+            return Parse(tokenScanner, new TokenMatcher(), this.AstBuilder);
+        }
+
+        public T Parse(TokenScanner tokenScanner, TokenMatcher tokenMatcher)
+        {
+            return Parse(tokenScanner, tokenMatcher, this.AstBuilder);

```

Comment:
Passing instance variables is my usual strategy when dealing with overloads. I'm happy to learn about alternative strategies.

In this concrete case, though: if you mean to remove the overload that takes an AstBuilder by "remove it from the next method" then of course the problem of having to deal with the overload disappears.